src/test/regress/output/multi_shard_modify.source

```diff
@@ -0,0 +1,249 @@
+-- commands with immutable functions in their quals
+CREATE FUNCTION temp_immutable_func() RETURNS integer AS 'SELECT 1737;' LANGUAGE SQL IMMUTABLE;
+SELECT master_multi_shard_modify('DELETE FROM customer_delete_protocol WHERE c_custkey = temp_immutable_func()');
+WARNING:  Bad result from localhost:57638
+DETAIL:  Remote message: function temp_immutable_func() does not exist
+ERROR:  Could not send query to shard placement.
+-- DELETE with expression in WHERE clause
+SELECT master_multi_shard_modify('DELETE FROM customer_delete_protocol WHERE c_custkey = (3*18)');
+ master_multi_shard_modify 
+---------------------------
+                         1
+(1 row)
+
+-- commands with a USING clause are unsupported
+CREATE TEMP TABLE temp_nations(name text, key integer);
+SELECT master_multi_shard_modify('DELETE FROM customer_delete_protocol USING temp_nations WHERE customer_delete_protocol.c_nationkey = temp_nations.key AND temp_nations.name = ''foobar'' ');
+ERROR:  cannot perform distributed planning for the given modification
+DETAIL:  Joins are not supported in distributed modifications.
+-- commands with a RETURNING clause are unsupported
+SELECT master_multi_shard_modify('DELETE FROM customer_delete_protocol WHERE c_custkey = 903 RETURNING *');
+ERROR:  cannot perform distributed planning for the given modification
+DETAIL:  RETURNING clauses are not supported in distributed modifications.
+-- commands containing a CTE are unsupported
+SELECT master_multi_shard_modify('WITH deleted_stuff AS (INSERT INTO customer_delete_protocol DEFAULT VALUES RETURNING *) DELETE FROM customer_delete_protocol');
+ERROR:  cannot perform distributed planning for the given modification
+DETAIL:  Common table expressions are not supported in distributed modifications.
+-- Check that we can successfully delete from multiple shards with 1PC
+SET citus.multi_shard_commit_protocol TO '1pc';
+SELECT count(*) FROM customer_delete_protocol;
+ count 
+-------
+  2999
+(1 row)
+
+SELECT master_multi_shard_modify('DELETE FROM customer_delete_protocol WHERE c_custkey < 100');
+ master_multi_shard_modify 
+---------------------------
+                        98
+(1 row)
+
+SELECT count(*) FROM customer_delete_protocol;
+ count 
+-------
+  2901
+(1 row)
+
+-- Check that we can successfully delete from multiple shards with 2PC
+SET citus.multi_shard_commit_protocol TO '2pc';
+SELECT count(*) FROM customer_delete_protocol;
+ count 
+-------
+  2901
+(1 row)
+
+SELECT master_multi_shard_modify('DELETE FROM customer_delete_protocol WHERE c_custkey < 200');
+ master_multi_shard_modify 
+---------------------------
+                       100
+(1 row)
+
+SELECT count(*) FROM customer_delete_protocol;
+ count 
+-------
+  2801
+(1 row)
+
+-- Check that shard pruning works
+SET client_min_messages TO DEBUG2;
+SELECT master_multi_shard_modify('DELETE FROM customer_delete_protocol WHERE c_custkey = 5001');
+DEBUG:  predicate pruning for shardId 102046
```

Comment:
We shouldn't have shard ID's in the output unless the shard id sequence is specifically set at the beginning.